synapse/streams/config.py

```diff
@@ -45,9 +46,13 @@ async def from_request(
         default_limit: int,
         default_dir: str = "f",
     ) -> "PaginationConfig":
-        direction = parse_string(
-            request, "dir", default=default_dir, allowed_values=["f", "b"]
+        direction_str = parse_string(
+            request,
+            "dir",
+            default=default_dir,
+            allowed_values=[Direction.FORWARDS.value, Direction.BACKWARDS.value],
         )
+        direction = Direction(direction_str)
```

Comment:
If we do this often we might want to make a `parse_string_to_enum` or something, but for this single instance it was easy enough...especially because this only has 2 values.